platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/config/internal/PlatformConfigUtils.java

```diff
@@ -102,4 +109,55 @@ private static Set<String> getConfigNames(@NonNull final Configuration configura
                 })
                 .collect(Collectors.toSet());
     }
+
+    /**
+     * Write all the settings to the file settingsUsed.txt, some of which might have been changed by settings.txt.
+     *
+     * @param directory the directory to write to
+     */
+    public static void writeSettingsUsed(final Path directory) {
+        final Configuration configuration = ConfigurationHolder.getInstance().get();
```

Comment:
Can we do this without using the config holder?

Reply:
This is used in `SignedStateFileWriter#writeSignedStateFilesToDirectory`. I intended to "hide" the use of `ConfigurationHolder` within that utility class for the time being. This is because one of the upcoming tasks is the removal of `ConfigurationHolder`. I decided to reserve that task for future Timo.